modules/hal_nordic/nrfx/CMakeLists.txt

```diff
@@ -101,7 +101,7 @@ zephyr_library_sources_ifdef(CONFIG_SOC_SERIES_NRF92X  ${MDK_DIR}/system_nrf92.c
 zephyr_library_sources(nrfx_glue.c)
 zephyr_library_sources(${HELPERS_DIR}/nrfx_flag32_allocator.c)
 zephyr_library_sources_ifdef(CONFIG_RETAINED_MEM_NRF_RAM_CTRL ${HELPERS_DIR}/nrfx_ram_ctrl.c)
-zephyr_library_sources_ifdef(CONFIG_NRFX_DPPI                 ${HELPERS_DIR}/nrfx_gppi_dppi.c)
+zephyr_library_sources_ifdef(CONFIG_NRFX_GPPI                 ${HELPERS_DIR}/nrfx_gppi_dppi.c)
```

Comment:
This does not seem to be necessary/beneficial.

Reply:
Can you elaborate? Do you mean that `CONFIG_NRFX_GPPI` kconfig is not beneficial or something else?

Reply:
`NRFX_GPPI` enables `NRFX_DPPI`, so the change is not really needed. But it causes that `nrfx_gppi_dppi.c` will be compiled also for `NRFX_PPI`, unnecessarily.

Reply:
It is possible to enable NRFX_DPPI without enabling any GPPI functionality now. Also NRFX_GPPI does more, as it is unified across the SoC families. Regardless of whether working with nRF52 or nRF54L or anything else, NRFX_GPPI will enable everything that's needed.

I just noticed that that `NRFX_PPI` in the next line should be replaced with `NRFX_GPPI` as well. I'll add this subsequent commit as well, if you don't mind.